packages/runner/src/lib/logger.js

```diff
@@ -2,6 +2,86 @@
 
 import _ from 'lodash'
 
+const _joinLogArgs = (args1, args2) => {
+
+  if (!_.isArray(args1)) {
+    return _joinLogArgs(_logCSS(args1, ''), args2)
+  }
+
+  if (!_.isArray(args2)) {
+    return _joinLogArgs(args1, _logCSS(args2, ''))
+  }
+
+  const ret1 = `${args1[0]} ${args2[0]}`
+  const ret2 = args1.slice(1).concat(args2.slice(1))
+
+  const toRet = [ret1].concat(ret2)
+
+  return toRet
+}
+
+// Fixes cases when string to style includes an escape character
+const _escapeConsoleString = (line) => {
+
+  // spacial sequences include '%c', '%f', '%d', etc
+  // can be escaped by doubling the amount of "%"
+  return line.replace(/(%+)(c|f|o|O|d|i|s)/g, '$1$1$2')
+
+}
+
+const _logColor = (line, color) => {
+  return _logCSS(line, `color:${color}`)
+}
+
+const _logCSS = (line, css) => {
+  return [`%c${_escapeConsoleString(line)}`, css]
+}
+
+const _formatDiffHeader = () => {
+  return ['Diff: ', _logColor('+ expected', 'green'), _logColor('- actual', 'red')].reduce(_joinLogArgs)
+}
+
+// this is tested in driver/assertions_spec
+const _formatConsoleDiff = (diff) => {
```

Comment:
there is only 1 test for this.

Reply:
yes I guess it is tested indirectly, but the logic is probably hairy enough to unit test it in isolation, since I believe this will be refactored at some point in the future and needs to be covered by its own tests, including all if-else branches